### ISSUE TITLE
Version 1 - Aim - Best Gurbani Reader, apart from Maharaj Saroop 🙏

### ISSUE THREAD

**Issue by cingh-jasdeep** [cingh-jasdeep/eGutkaSahib-app#3]
- controls bar
- font size adjustment
- paragraph view (useful for being used along with kirtan, so that full shabad can be viewed once)
- line by line view, with translations, transliterations etc. (for sehaj paath)
- full transliteration mode (english, hindi, shahmukhi, for readers in other languages)
- static page navigation (no sliding) (useful for e-ink screens and elder users)
- individual word meanings for Gurmukhi, from (check individual word, or line word definitions possible in db) (useful for Sehaj Paath)
- bani list with bani's view (maybe need baani length also, as per preference, would like inside each baani and help user choose one)
- need e-ink display friendly ui
- beautiful border and design
- kirtanee view, should support front back navigation history, to go back to previous shabad/bani (asi di vaar)
- sehaj paath view, easy to find meanings, full immersion with lareevar, assist available when needed for meanings and padched, 
- nitnem paath, easy available to adjust baani lengths (eg. rehraas sahib) within each baani, easy similar to sehaj paath other features, but more accustomed to chalant reading, can include bookmarks to different sections of baani like sundar gutka

for feedback, need a system, 
if many sangat ask  for a feature update may be able to push that update, to physical devices also,
with minimal fuss
could have many tech features but priority is easy ux, as many elderly would likely adjust rather than complain
-eyes friendly, (exclusive software features for app only)
-satkaar friendly, (beautiful hardware covers)
-elderly friendly, (font size and ui in different languages)

Prototype creation

1st would be 
- software
as only asset currently is time
software itself would be a good end product

then hardware
- till some time when financial is stable,
can order a machine online and make prototype

beautiful machine needed

okay then let's begin,
see if hardware available easily, if not let's start

let's begin software, which platfrom? flutter as it is availble on web, android, and embedded

Basically 

- kirtanee friendly, 
- sehaj paath friendly, 
- nitnem paath friendly, 
- eyes friendly, 
- satkaar friendly
- elderly friendly

**Comment by cingh-jasdeep** [comment]
question is whether translations/transliterations are only available in continuous reading?
or only available on long press?
continous reading
pros 
could help see meaning in continous reading
maybe could set as option, or could totally avoid it
avoiding it for now, as it could help in better Gurbani reading experience

idea is get assistance when you need it
but try to stick to pure Gurbani whenever possible

this is current stand point

**Comment by cingh-jasdeep** [comment]
question next page turn on click on either side of page or show menu and then next page turn?
(navigation needed in only when baani overflows the page)
kindle 
use touch region based page navigation
easier to have more text on screen
but may not be elderly friendly,

 one option can be 2 taps on either side of page horizontally, by showing small arrows, on single tap, (can also include support for double tap navigation, without showing menu
which is naturally inferred from the ui, thanks Babaji :) )
this would show menu also, satisfy full screen readers as everything would go away on another tap anywhere on page that does not have any controls,

on long tap we would have Gurbani translations/transliterations options.
this solutions seems
- elderly friendly
- kirtanee friendly, 
- sehaj paath friendly,
- nitnem paath friendly, 
- eyes friendly, 
- Gurbani satkaar friendly,

**Comment by cingh-jasdeep** [comment]
> question is whether translations/transliterations are only available in continuous reading? or only available on long press? continous reading pros could help see meaning in continous reading maybe could set as option, or could totally avoid it avoiding it for now, as it could help in better Gurbani reading experience
> 
> idea is get assistance when you need it but try to stick to pure Gurbani whenever possible
> 
> this is current stand point
> 
> can change according to Sangat's demand

one problem could be for english/spanish users, for whom continous meanings might be a neccessity, as they would need meanings intitally to help understand the vastly different language.
therefore settings should be available, but could be turned off by default to make it appropriate for most use cases.
location should be inside common text options and can be made available on additional languages settings.

**Comment by cingh-jasdeep** [comment]
- kirtanee friendly, 
- sehaj paath friendly, 
- nitnem paath friendly, 
- eyes friendly, 
- satkaar friendly, 
- elderly friendly
- multilingual friendly (inclusive)

dark mode?
in main actions or settings page?
for e-ink app also may need.

main actions elderly friendly, more clutter may be not be satkaar friendly, , (as not main option, as mainly gurbani is on white only, sehaj,) definately eyes friendly (may not be in e-ink but def in mobile apps), elderly friendly def, multilingual friendly (yes)

may be not be satkaar friendly, but essentially we are making a reader, people can do it from other apps anyways, or settings also, that won't be elderly friendly, i think best

for mobile apps this should be in actions, as screens become much more eyes friendly.
for e-ink we can put this in settings. but regardless we may need a settings icon as well.

**Comment by cingh-jasdeep** [comment]
for continuous update for app, from server, flutter might not be best choice.
need this for being elderly friendly, also e-reader may have this option.
may still be possible in flutter through app store and solve elderly friendly.
can be researched furthur.

**Comment by cingh-jasdeep** [comment]
mockups started here
https://github.com/Gurbani-Is-Jag-Meh-Chaanan/eGutkaSahib-app/issues/4#issue-1186498136
, shall post updates in it's own issue

**Comment by cingh-jasdeep** [comment]
Colors themes:

Themes tailored to color displays
- color_light
- color_dark

Themes tailored to black and white eink displays
- bw_light
- bw_dark


main colors:
need to follow material color guideline to construct these colors

**Comment by cingh-jasdeep** [comment]
motivation tips in each sections:

- kirtanee friendly,
- sehaj paath friendly,
- nitnem paath friendly,
- eyes friendly,
- satkaar friendly,
- elderly friendly
- multilingual friendly (inclusive)

eg.
- kirtanee friendly,
gaviai suniai man rakhiai bhao dukh parhar sukh ghar lai jaye

- sehaj paath friendly
pothi parmeshar ka thaan

- nitnem paath friendly
nit uth gaavo prabh ki baani,
gavoh suno paro nit bhai,

good but opiniated to baani
can get more motivation tuks from other people like Giani ji.
would like good :)

but have to be very careful not to misinterpret as many mahapurakhs have different meanings.
maybe can keep as is for now.
advanced level.

**Comment by cingh-jasdeep** [comment]
- kirtanee friendly,
- sehaj paath friendly,
- nitnem paath friendly,
- eyes friendly,
- satkaar friendly,
- elderly friendly
- multilingual friendly (inclusive)

question is 
on long press gurbani 
whether to show all translations/transliterations or 
show only selected by reader

only



also, whether to give reader option 
to select if to show translation/transliteration alongside baani or on long press or both (default) 
multilingual friendly okay but could be a nuisance to see only one line meaning out of context
would prefer word meaning.
aim is to have reader read fluently and intuitively understand and love gurbani without getting confused in meanings


solution could be to have everywhere by default if the reader took the pain to turn it on in the first place
-can give option on long press if reader does not want on long press

problem is word meanings might not available but we can make whatever we have first.

**Comment by cingh-jasdeep** [comment]
kirtanee friendly,
sehaj paath friendly,
nitnem paath friendly,
eyes friendly,
satkaar friendly,
elderly friendly
multilingual friendly (inclusive)

question is 
which db to use

till now we have seen following features


| Db/Feature  | ShabadOS |  BaniDB |
| ------------- | ------------- | ------------- |
| Amrit Keertan | not available in stable (need more check) | maybe (in sttm.org) |
| Shabad Kosh (punjabi)  | not available in stable (need more check) | maybe (in sttm.org) |

**Comment by cingh-jasdeep** [comment]
kirtanee friendly,
sehaj paath friendly,
nitnem paath friendly,
eyes friendly,
satkaar friendly,
elderly friendly
multilingual friendly (inclusive)

which front end framework to use

Framework/Feature | KOReader | Flutter | Ionic Vue 
|-----|------|------|------
performance | great | great | good
custom ui | full custom | custom | custom with Tailwindcss
components | available | common available | common available
dev setup | no experience | have experience | no experience

**Comment by cingh-jasdeep** [comment]
notes left in project repo
https://github.com/Gurbani-Is-Jag-Meh-Chaanan/eGutkaSahib-project/blob/main/FlutterStateManagementNotes.md

**Comment by cingh-jasdeep** [comment]
Flutter research to 
make it e-ink friendly

- no sliding
- no animations
- no shadows
- simple colors

**Comment by cingh-jasdeep** [comment]
hardware review:

Supplier/Reviews | [Onyx/boox](https://onyx.en.alibaba.com/?spm=a2700.12243863.0.0.68813e5f7GsoiK)
----|----
Min Price | 125 USD

**Comment by cingh-jasdeep** [comment]
> kirtanee friendly, sehaj paath friendly, nitnem paath friendly, eyes friendly, satkaar friendly, elderly friendly multilingual friendly (inclusive)
> 
> hardware review:
> 
> Supplier/Reviews	[Onyx/boox](https://onyx.en.alibaba.com/?spm=a2700.12243863.0.0.68813e5f7GsoiK)
> Min Price	125 USD

got message from supplier
need to check if it would be compatible with flutter.
price range is good.
need to ask for cases.

**Comment by cingh-jasdeep** [comment]
draft: reply to shenzen diyan
Hello Candy,

Thanks for your reply.

Can you kindly confirm the following features for this device?

 

- Does this device have a touch screen?

- Does this device support custom software? if yes, can you give a link to SDK for application development?

- What is the expected battery life of this device? (for example, does it last 2 weeks when used for 2 hours daily?)

- Does this device support a protective case accessory? if yes, is it possible to have a custom design for the protective case as well?

**Comment by cingh-jasdeep** [comment]
started software notes in #6

**Comment by cingh-jasdeep** [comment]
understanding kickstarter and it's complexities of hardware manufacturing.

as hardware specifications can help build a better software product
but fortunately that device had arm on it.
so work in flutter is not waste but it's good to know what you shall be facing!
babaji, let this be your work and let my intellect follow.

**Comment by cingh-jasdeep** [comment]
started hardware research #7

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

Researched database options,
queries, overflow page view widget, 
#6 

Updated mock for "Open Shabad In"
#4

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

Explored options for database
#6 

started translation keys for
all labels
https://github.com/Gurbani-Is-Jag-Meh-Chaanan/eGutkaSahib-app/tree/master/assets/locales

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

Continued with translation keys for
all labels
https://github.com/Gurbani-Is-Jag-Meh-Chaanan/eGutkaSahib-app/tree/master/assets/locales

Updated mock diagram 


Researched Gurbani DB more
Research Gurbani Fonts, Gurmukhi-Utils for labels,

Waheguru Ji Mehar Banayi Rakhna
Manukha Janam safal kara lena jio
Ghaal khaye kara laina,
cheet nirnjan naal kara laina,
apni bhagti la laina,
shabad naal jor laina ji,
